### PR TITLE
NPM Package Workflows

### PR DESCRIPTION
**Pull request by jaredtbrown**
# Overview
This will add 2 new workflows:
1. To build and test a package
1. To publish a package

## testing
I setup the encompass-proxy-client pr workflow to point to this new workflow for testing.

## todo
Things I would like to see done in the future:
- Validate version number changed to something that does not already exist.

### CODE REVIEW

**Comment by tommyk** [comment]
where is the part that uploads the npm package to a npm registry?

IE. `npm publish`

File: .github/workflows/typescript-build-and-test-package.yml [hunk 1]
```diff
@@ -24,7 +24,7 @@ jobs:
         run: yarn install --production
 
       - name: "build artifact"
-        run: zip -r build.zip build package.json
+        run: zip -r build.zip build package.json node_modules
```
Comment (tommyk) [inline .github/workflows/typescript-build-and-test-package.yml]: why are we adding the `node_modules` folder for an npm package?

Comment (tommyk) [reply .github/workflows/typescript-build-and-test-package.yml]: honestly, why even save an artifact if we are uploading to an npm package? as that is the artifact right? thoughts?

Comment (jaredtbrown) [reply .github/workflows/typescript-build-and-test-package.yml]: We probably don't need them. I don't remember why I added that as I originally had it without.

But the artifact here was just going to be used to publish in another job. But actually, I don't think we need to do that at all.

Comment (tommyk) [reply .github/workflows/typescript-build-and-test-package.yml]: when building a zip for a lambda, the `node_modules` has to be there.  for this, it won't end up in the npm package, and the build artifact isn't really usable either.

Comment (jaredtbrown) [reply .github/workflows/typescript-build-and-test-package.yml]: Alright, I removed all the stuff with creating an artifact. It should now just run the build and test scripts

**Comment by jaredtbrown** [comment]
Marked as a draft so I can experiment with validating the package version number

File: .github/workflows/build-and-test-npm-package.yml [hunk 2]
```diff
@@ -0,0 +1,39 @@
...
+      github_token:
+        required: true
+
+jobs:
+  build_and_test:
+    runs-on: ubuntu-latest
+    env:
+      NPM_TOKEN: ${{ secrets.npm_token }}
+      GITHUB_TOKEN: ${{ secrets.github_token }}
+      PACKAGE_VERSION_EXISTS: 0
```
Comment (tommyk) [inline .github/workflows/build-and-test-npm-package.yml]: i think this is left over?

**Review by tommyk — APPROVED** [review]

File: .github/workflows/build-and-test-npm-package.yml (hunk 2 above)
Comment (jaredtbrown) [reply .github/workflows/build-and-test-npm-package.yml]: Yep. Removed

**Pull request merged**